[CORE-9244] `rptest`: deflake `datalake/compaction_test.py`

WillemKauf opened the pull request:
This test should have had it's `min.cleanable.dirty.ratio` set to 0.0 in commit https://github.com/redpanda-data/redpanda/pull/24991/commits/cba07b87f82bce719b33745391b0c02efd1c9e80 for unconditional compacting of the log per expectations of the test.

Also, alter some backoff and timeout parameters for additional leniency.

## Backports Required

- [X] none - not a bug fix
- [ ] none - this is a backport
- [ ] none - issue does not exist in previous branches
- [ ] none - papercut/not impactful enough to backport
- [ ] v24.3.x
- [ ] v24.2.x
- [ ] v24.1.x

## Release Notes

* none

WillemKauf committed:
`rptest`: deflake `datalake/compaction_test.py`

This test should have had it's `min.cleanable.dirty.ratio`
set to 0.0 for unconditional compacting of the log per
expectations of the test.

Also, alter some backoff and timeout parameters for additional
leniency.

vbotbuildovich commented:
<!--TEST RESULTS-->
<!--9bb3cd91-5e4f-4635-8725-1bd17da6e9f8-->
<!--TEST RESULTS-->
<!--7a03e61d-a482-4bc7-a0f6-1a0d4e1fe367-->
<!--TEST RESULTS-->
<!--f773111f-edbc-4f37-882d-e3bc853054fc-->
<!--title-->
## CI test results
<!--/title-->
<!--body-->
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62499">build#62499</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| gtest_raft_rpunit.gtest_raft_rpunit | unit | https://buildkite.com/redpanda/redpanda/builds/62499#01954fde-baa3-4300-bfd9-a465b9a8ebcc | FLAKY | 1/2 |
| rptest.tests.scaling_up_test.ScalingUpTest.test_scaling_up_with_recovered_topic | ducktape | https://buildkite.com/redpanda/redpanda/builds/62499#01955026-9ff0-4a25-b8d0-3a79bde56d7d | FLAKY | 1/2 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62506">build#62506</a></summary>


| test_id | test_kind | job_url | test_status | passed |
|:--------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_HADOOP | ducktape | https://buildkite.com/redpanda/redpanda/builds/62506#019551a4-e9f9-4f2d-ba47-2a9a530bf1a7 | FLAKY | 20/22 |
| rptest.tests.datalake.compaction_test.CompactionGapsTest.test_translation_no_gaps.cloud_storage_type=CloudStorageType.S3.catalog_type=CatalogType.REST_JDBC | ducktape | https://buildkite.com/redpanda/redpanda/builds/62506#019551a4-e9fb-4947-8f6e-734bd2f0b3a3 | FLAKY | 20/21 |
</details>
<details>
<summary>test results on <a href="https://buildkite.com/redpanda/redpanda/builds/62509">build#62509</a></summary>


| test_id                                                                                                                                                                                                                                           | test_kind   | job_url                                                                                   | test_status   | passed   |
|:--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|:------------|:------------------------------------------------------------------------------------------|:--------------|:---------|
| rptest.tests.cloud_storage_timing_stress_test.CloudStorageTimingStressTest.test_cloud_storage.cleanup_policy=compact.delete                                                                                                                       | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62509#0195524e-bc35-4298-b7cd-0c9d4928f14a | FLAKY         | 1/2      |
| rptest.tests.compaction_recovery_test.CompactionRecoveryUpgradeTest.test_index_recovery_after_upgrade                                                                                                                                             | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62509#0195524e-bc36-43fd-9c3d-8f29f2033656 | FLAKY         | 1/2      |
| rptest.tests.data_migrations_api_test.DataMigrationsApiTest.test_creating_and_listing_migrations                                                                                                                                                  | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62509#01955234-4294-42d1-a154-c1a10ac19a33 | FLAKY         | 1/2      |
| rptest.tests.full_disk_test.LogStorageMaxSizeSI.test_stay_below_target_size.log_segment_size=1048576.cleanup_policy=compact                                                                                                                       | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62509#01955234-4294-42d1-a154-c1a10ac19a33 | FLAKY         | 1/2      |
| rptest.tests.random_node_operations_test.RandomNodeOperationsTest.test_node_operations.enable_failures=False.mixed_versions=False.with_tiered_storage=True.with_iceberg=False.with_chunked_compaction=True.cloud_storage_type=CloudStorageType.S3 | ducktape    | https://buildkite.com/redpanda/redpanda/builds/62509#0195524e-bc36-43fd-9c3d-8f29f2033656 | FLAKY         | 1/2      |
</details>
<!--/body-->

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/datalake/compaction_test.py

WillemKauf committed:
`rptest`: deflake `CompactionGapsTest`

This test was still having trouble reaching the desired number
of segments.

Alter the `log_compaction_interval_ms` and the timeout/backoff values
to give the test more leniency in reaching the desired number of segments
produced.

WillemKauf commented:
/ci-repeat 10
skip-redpanda-build
skip-units
dt-repeat=20
tests/rptest/tests/datalake/compaction_test.py

WillemKauf commented:
No flakes in [Build #62511](https://buildkite.com/redpanda/redpanda/builds/62511).
